File: 프로그래머스/lv2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
	counter={}
	for i in tangerine:
		if i in counter:
			counter[i]+=1
		else:
			counter[i]=1

	sorted_counter=sorted(counter.items(), key=lambda item:item[1],reverse=True)
	array=[]
	for i in sorted_counter:
		for _ in range(i[1]):
			array.append(i[0])
	
	ans=array[:k]
	ans=set(ans)
	return len(ans)
	
	

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        HashMap<Integer, Integer> counter = new HashMap<>();
        for(int i : tangerine){
            if(counter.containsKey(i)){
                counter.put(i,counter.get(i)+1);
            }else{
                counter.put(i,1);
            }
        }
        
        List<Map.Entry<Integer, Integer>> sortedCounter = new ArrayList<>(counter.entrySet()); // entry set으로 list로 만든다. 
        
        sortedCounter.sort((a, b) -> b.getValue().compareTo(a.getValue())); 
        // value값으로 비교해서 정렬
        List<Integer> array = new ArrayList<>();
        
        for (Map.Entry<Integer, Integer> entry : sortedCounter) {
            for (int i = 0; i < entry.getValue(); i++) {
                array.add(entry.getKey());
            }
        }

        Set<Integer> answer = new HashSet<>(array.subList(0, k));
        return answer.size();
        
    
    }
}